gtk/gtk.list-box.lisp

```diff
@@ -0,0 +1,585 @@
+
+#+gtk-3-10
+(defcfun gtk-list-box-select-all :void
+  (box (g-object gtk-list-box)))
+
+#+gtk-3-10
+(export 'gtk-list-box-select-all)
+
+;;; ----------------------------------------------------------------------------
+;;; gtk_list_box_unselect_all ()
+;;; ----------------------------------------------------------------------------
+
+#+gtk-3-10
+(defcfun gtk-list-box-unselect-all :void
+  (box (g-object gtk-list-box)))
+
+#+gtk-3-10
+(export 'gtk-list-box-unselect-all)
+
+;;; ----------------------------------------------------------------------------
+;;; gtk_list_box_get_selected_row ()
+;;; ----------------------------------------------------------------------------
+
+#+gtk-3-10
+(defcfun gtk-list-box-get-selected-row (g-object gtk-list-box-row)
+  (box (g-object gtk-list-box)))
+
+#+gtk-3-10
+(export 'gtk-list-box-get-selected-row)
+
+;;; ----------------------------------------------------------------------------
+;;; gtk_list_box_selected_foreach ()
+;;; ----------------------------------------------------------------------------
+
+#+gtk-3-14
+(defcallback gtk-list-box-foreach-func-callback :boolean
+    ((box (g-object gtk-list-box))
+     (row (g-object gtk-list-box-row))
+     (data :pointer))
+  (restart-case
+      (funcall (glib::get-stable-pointer-value data) box row)
+    (return () NIL)))
+
+#+gtk-3-14
+(defcfun ("gtk_list_box_selected_foreach" %gtk-list-box-selected-foreach) :void
+  (box (g-object gtk-list-box))
+  (func :pointer)
+  (data :pointer))
+
+#+gtk-3-14
+(defun gtk-list-box-selected-foreach (box func)
+  (with-stable-pointer (ptr func)
+    (%gtk-list-box-selected-foreach box (callback gtk-list-box-selected-foreach-callback) ptr)))
+
+#+gtk-3-14
+(export 'gtk-list-box-selected-foreach)
+
+;;; ----------------------------------------------------------------------------
+;;; gtk_list_box_get_selected_rows ()
+;;; ----------------------------------------------------------------------------
+
+#+gtk-3-14
+(defcfun gtk-list-box-get-selected-rows (g-list (g-object gtk-list-box-row))
+  (box (g-object gtk-list-box)))
+
+#+gtk-3-14
+(export 'gtk-list-box-get-selected-rows)
+
+;;; ----------------------------------------------------------------------------
+;;; gtk_list_box_get_adjustment ()
+;;; ----------------------------------------------------------------------------
+
+#+gtk-3-10
+(defcfun ("gtk_list_box_get_adjustment" gtk-list-box-adjustment)
+    (g-object gtk-adjustment)
+  (box (g-object gtk-list-box)))
+
+;;; ----------------------------------------------------------------------------
+;;; gtk_list_box_set_adjustment ()
+;;; ----------------------------------------------------------------------------
+
+#+gtk-3-10
+(defcfun ("gtk_list_box_set_adjustment" %gtk-list-box-set-adjustment)
+    :void
+  (list-box (g-object gtk-list-box))
+  (adjustment (g-object gtk-adjustment)))
+
+#+gtk-3-10
+(defun (setf gtk-list-box-adjustment) (new-value list-box)
```

Comment:
Unless I'm mistaken, this is inconsistent with the rest of the library: we usually don't use `setf` expanders for methods that are not property writers.